### ISSUE TITLE
Add a documentation page on error handling

### ISSUE THREAD

**Issue by Daenyth** [typelevel/cats#3523]
There's some major gaps around error handling in the official docs, especially painful because the whole ecosystem relies on them - library users are often instructed to use MonadError/ApplicativeError error handling code, but it's not present in the documentation

- Add a page on ApplicativeError/MonadError (one page, likely)
  - Either on that page or on its own page, provide a "cookbook" of common error handling examples

This is mentioned in #1801, but I thought it was worth calling out as its own issue

**Comment by dhinojosa** [comment]
Can I take this? I had been doing some trials on this, and this would help me understand them a bit better @Daenyth @barambani

**Comment by Daenyth** [comment]
@dhinojosa Absolutely! Please hop on gitter and ask around if you run into anything you want help with while working on it. We're always very happy to have new contributors, and documentation makes the library better for all users :)

**Comment by barambani** [comment]
brilliant @dhinojosa , thanks for doing it

**Comment by dhinojosa** [comment]
Thanks all, I have been meaning to add and join for a long time

**Comment by dhinojosa** [comment]
`ApplicativeError` is up for approval. #3593

**Comment by dhinojosa** [comment]
Waiting until the mdoc #3595 is approved and I will write the docs in mdoc.

**Comment by dhinojosa** [comment]
PR up for review.
https://github.com/typelevel/cats/pull/3616

**Comment by dhinojosa** [comment]
PR up for review for `MonadError`, let me know what you think @Daenyth. I haven't added your tupling recommendation yet that you suggested in the previous PR.

#3639
